iroh/tests/provide.rs

```diff
@@ -716,3 +727,106 @@ async fn test_custom_request_collection() {
     .expect("timeout")
     .expect("get failed");
 }
+
+#[derive(Clone, Debug)]
+struct CustomAuthHandler;
+
+impl RequestAuthorizationHandler for CustomAuthHandler {
+    fn authorize(
+        &self,
+        _db: Database,
+        token: Option<RequestToken>,
+        _request: &iroh_bytes::protocol::Request,
+    ) -> BoxFuture<'static, anyhow::Result<()>> {
+        async move {
+            match token {
+                Some(token) => {
+                    if token.to_bytes() != vec![1, 2, 3, 4, 5, 6] {
+                        anyhow::bail!("bad token")
+                    }
+                    Ok(())
+                }
+                None => {
+                    anyhow::bail!("give token plz")
+                }
+            }
+        }
+        .boxed()
+    }
+}
+
+#[tokio::test]
+async fn test_token_passthrough() {
+    let rt = test_runtime();
+    let readme = readme_path();
+    let (db, hash) = create_collection(vec![readme.into()]).await.unwrap();
+    let provider = match Node::builder(db)
+        .bind_addr("[::1]:0".parse().unwrap())
+        .custom_auth_handler(CustomAuthHandler)
+        .runtime(&rt)
+        .spawn()
+        .await
+    {
+        Ok(provider) => provider,
+        Err(_) => {
+            // We assume the problem here is IPv6 on this host.  If the problem is
+            // not IPv6 then other tests will also fail.
+            return;
+        }
+    };
+
+    let token = Some(RequestToken::from(vec![1, 2, 3, 4, 5, 6]));
+    let mut events = provider.subscribe();
+    let (tx, rx) = tokio::sync::oneshot::channel();
+    tokio::spawn(async move {
+        let mut token = None;
+        while let Ok(msg) = events.recv().await {
+            match msg {
+                Event::ByteProvide(bp_msg) => {
+                    if let iroh_bytes::provider::Event::GetRequestReceived { token: tok, .. } =
+                        bp_msg
+                    {
+                        // println!("token: {:?}", token);
+                        token = tok;
+                        break;
+                    }
+                }
+            }
+        }
+        tx.send(token).unwrap();
```

Comment:
Instead of the oneshot, just return the value at the end, and await the handle returned from `tokio::spawn`.